banzai_nres/dbs.py

```diff
@@ -69,6 +70,37 @@ def diff_luminosity(cls, value):
         return func.abs(cls.luminosity - value)
 
 
+# We define the great circle distance here instead of using astropy because we need it to work inside the db.
+def cos_great_circle_distance(sin_ra1, cos_ra1, sin_dec1, cos_dec1, sin_ra2, cos_ra2, sin_dec2, cos_dec2):
+    cos_distance = sin_dec1 * sin_dec2 + cos_dec1 * cos_dec2 * (cos_ra1 * cos_ra2 + sin_ra1 * sin_ra2)
```

Comment:
minor comment to add a reference to where this formula is from. For instance, this looks different than https://undergroundmathematics.org/trigonometry-compound-angles/the-great-circle-distance

Reply:
It's just the standard great circle distance formula that is on e.g. wikipedia, but I added some docs.